scaaml/capture/input_generators/attack_point_iterator.py

```diff
@@ -238,3 +239,48 @@ def __iter__(self) -> AttackPointIteratorT:
 
     def get_generated_keys(self) -> List[str]:
         return self._configuration_iterator.get_generated_keys()
+
+
+class AttackPointIteratorZip(AttackPointIterator):
+    """Attack point iterator zip class. This class takes any amount of operands
+    and combines them just like the zip function in python."""
+
+    def __init__(self, operation: str, operands: List[Dict[str, Any]]) -> None:
+        """Initialize the zip iterate.
+          
+          Args:
+            operation (str): The operation of the iterator
+                represents what the iterator does and what 
+                has to be in the config file. This is only used once to
+                double check if the operation is the correct one.
+                
+            operands (List[Dict[str, Any]]): The operands are any number of
+                iterator configs that will be combined."""
+        assert "zip" == operation
+        self._operands = list(
+            build_attack_points_iterator(operand) for operand in operands)
+        self._len = 0
+        for operand in self._operands:
+            try:
+                if self._len > len(operand) or self._len <= 0:
+                    self._len = len(operand)
+            except LengthIsInfiniteException:
+                self._len = -1
+
+    def __len__(self) -> int:
+        if self._len < 0:
+            raise LengthIsInfiniteException("The length is infinite!")
+        return self._len
+
+    def __iter__(self) -> AttackPointIteratorT:
+        return iter(
```

Comment:
Honestly I had quite hard time reading this (three for loops in an iterator expression is too many for me). 
WDYT about:

```
# zip will return a tuple of output dictionaries, we flatten them.
return iter(merge_dictionaries(tuple_of_dictionaries) for tuple_of_dictionaries in zip(*self._operands))
```

With a function `merge_dictionaries` which gets a tuple of dictionaries and uses `update` method of a Python dictionary to merge them (similar to `get_generated_keys` below). This way we can easily unit-test the `merge_dictionaries`.